aws/resource_aws_instance.go

```diff
@@ -1341,6 +1341,21 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 				if err != nil {
 					return fmt.Errorf("Error updating metadata options: %s", err)
 				}
+				stateConf := &resource.StateChangeConf{
+					Pending:    []string{ec2.InstanceMetadataOptionsStatePending},
+					Target:     []string{ec2.InstanceMetadataOptionsStateApplied, ec2.InstanceStateNameRunning},
```

Comment:
API docs indicate "applied" state follows the "pending" state returned from `ModifyInstanceMetadataOptions`  but in testing this, "applied" wasn't recognized as a valid target state, only "running" ..perhaps this should just be kept as "running"?

Reply:
would we consider moving these statechangeconf references to a service package like the ec2 waiter/status ones as part of this tech-debt item? looking at the resource, we have quite a number of references that maybe could be consolidated

Reply:
We might need a separate refresh function for this as those enumerations appear to be related to the `InstanceMetadataOptionsResponse` type:

```go
// The metadata options for the instance.
type InstanceMetadataOptionsResponse struct {
// ...
	// The state of the metadata option changes.
	//
	// pending - The metadata options are being updated and the instance is not
	// ready to process metadata traffic with the new selection.
	//
	// applied - The metadata options have been successfully applied on the instance.
	State *string `locationName:"state" type:"string" enum:"InstanceMetadataOptionsState"`
}
```

Which can be found in the `Instance.MetadataOptions` field. 👍 

If we are adding new functions, I think adding them to the internal package is fine and we can worry about refactoring existing ones separately.

Reply:
As for the block device changes, there's also a `Status` field buried deep in `Instance.BlockDeviceMappings[].Ebs.Status`, but not sure how well that aligns with things like updating delete on termination and such, as its enumeration is:

```go
func AttachmentStatus_Values() []string {
	return []string{
		AttachmentStatusAttaching,
		AttachmentStatusAttached,
		AttachmentStatusDetaching,
		AttachmentStatusDetached,
	}
}
```

Reply:
@bflad the enumerations here didn't seem to work as we'd hope so instead I'm experimenting with an updated refresh func in 9f46770 to check for the actual `deleteOnTermination` value. would that be more in line with what we this type of update?